packages/block-library/src/navigation/edit/index.js

```diff
@@ -305,7 +305,8 @@ function Navigation( {
 				if ( a.locations < b.locations ) {
 					return 1;
 				}
-				if ( a.locations > b.locations ) {
+				// If no locations are assigned, we compare the term_id to determine and sort by the most recently created.
+				else if ( a.locations > b.locations || a.term_id < b.term_id ) {
```

Comment:
Why are locations in this part of the calculation?

Reply:
Good question, I realized the logic was convoluted. Cleaned this up in https://github.com/WordPress/gutenberg/pull/46286/commits/9abfcd52d455777c09aebeeefaea06ada4d2bb5c